src/main/java/urjc/isi/controladores/PeliculasController.java

```diff
@@ -89,13 +89,129 @@ public static String selectAllPeliculas(Request request, Response response) thro
+		} else if(request.queryParams("director")!= null) {
+			output = ps.getRankingByDirector(request.queryParams("director"));
+			result = result + "Peliculas en las que participa " + request.queryParams("director") + "<br/><br/>";
+		} else if(request.queryParams("guionista")!= null) {
+			output = ps.getRankingByGuionista(request.queryParams("guionista"));
+			result = result + "Peliculas en las que participa " + request.queryParams("guionista") + "<br/><br/>";
+		} else if(request.queryParams("genero")!= null) {
+			output = ps.getRankingByGenero(request.queryParams("genero"));
+			result = result + "Peliculas en las que participa " + request.queryParams("genero") + "<br/><br/>";
+		} else {
+			output = ps.getBestPeliculas();
+			result = result + "10 peliculas mejor valoradas" + "<br/><br/>";
+		}
+		
+		if(request.queryParams("format")!= null && request.queryParams("format").equals("json")) {
+			response.type("application/json");
+			JsonObject json = new JsonObject();
+			json.addProperty("status", "SUCCESS");
+			json.addProperty("serviceMessage", "La peticion se manejo adecuadamente");
+			JsonArray array = new JsonArray();
+			for(int i = 0; i < output.size(); i++) {
+				array.add(output.get(i).toJSONObject());;
+			}
+			json.add("output", array);
+			result = json.toString();
+		} else {	
+			for(int i = 0; i < output.size(); i++) {
+			    result = result + output.get(i).toHTMLString() +"</br>";
+			}
+		}
+		return result;		
+	} 
+	
+	//Devuelve peliculas en funcion de la calificacion
+	
+	public static String calificacion(Request request, Response response) throws SQLException {
+		
+		List<Peliculas> output;
+		
+		String result = "<form action='/peliculas/calificacion' method='get' enctype='multipart/form-data'>"
+						+ "<button type=submit name=adultos value=adultos>Adultos </button><br/></form>"
+						+ "<form action='/peliculas/calificacion' method='get' enctype='multipart/form-data'>"
+						+ "<button type=submit name=ninos value=ninos>Niños </button><br/></form>"
+						+ "Introduce una pelicula: <br/><br/>"
+						+ "<form action='/peliculas/calificacion' method='get' enctype='multipart/form-data'>"
+						+ "Pelicula: <input type=text name=pelicula size=30>"
+						+ "<button type=submit value=Pelicula>Buscar </button><br/></form>"
+						+ "<br/>"
+						+ "<form action='/peliculas/calificacion' method='get' enctype='multipart/form-data'>"
+						+ "<button type=submit value=VolverAtrás>Volver atrás </button>"
+						+ "<br/><br/>";
+		
+		if(request.queryParams("adultos") != null) {
+			output = ps.getAllPeliculasForAdultos();
+			result = result + "Peliculas para adultos " + "<br/><br/>";
+		} else if(request.queryParams("ninos") != null) {
+			output = ps.getAllPeliculasForNinos();
+			result = result + "Peliculas para niños " + "<br/><br/>";
+		} else if(request.queryParams("peliculas") != null) {
+			output = ps.getCalificacionForPelicula(request.queryParams("pelicula"));
```

Comment:
Miráis si películas está vació pero luego introducís película (en singular) en el método del service. Seguramente sea ésto. Aun así, posteriormente os devuelve una lista de películas, cuando el método solo deberíais decir si es adulta o no, entiendo.

Reply:
Volvemos también a la pregunta de que, si ya tenemos un método que lista todas las películas, podría ser usada para filtrarlas también si son adultas o no en vuestra funcionalidad, porque ya que tenéis las querys, podéis directamente colocarlas también como un filtro de selectAll.